napari/components/layerlist.py

```diff
@@ -54,6 +55,20 @@ def _on_selection_changed(self, event):
         for layer in event.removed:
             layer._on_selection(False)
 
+    def _process_delete_item(self, item: Layer):
+        item.events.set_data.disconnect(self._clean_cache)
+        self._clean_cache()
+
+    def _clean_cache(self):
+        if 'extent' in self.__dict__:
+            del self.extent
+        if "_extent_world" in self.__dict__:
+            del self._extent_world
+        if "_step_size" in self.__dict__:
+            del self._step_size
+        if "_ranges" in self.__dict__:
+            del self._ranges
```

Comment:
What do you think of a compact notation like the following?:

```suggestion
        cached_properties = ('extent', '_extent_world', '_step_size',
                             '_ranges')
        [self.__dict__.pop(p, None) for p in cached_properties]

```

Reply:
it looks better. Let's check if it works as expected.